modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/redshift/internal/ShredModelEntry.scala

```diff
@@ -0,0 +1,186 @@
+package com.snowplowanalytics.iglu.schemaddl.redshift.internal
+
+import cats.Show
+import cats.syntax.show._
+import com.snowplowanalytics.iglu.schemaddl.jsonschema.Pointer.SchemaPointer
+import com.snowplowanalytics.iglu.schemaddl.jsonschema.{Pointer, Schema}
+import ColumnTypeSuggestions.columnTypeSuggestions
+import io.circe.{ACursor, Json}
+
+import scala.annotation.tailrec
+
+/**
+ * Single bottom level entry of the schema model. Each entry matches a single column in warehouse.
+ *
+ * @param ptr       - json pointer. A cursor that could be used to extract the data from json event.
+ * @param subSchema - jsonschema of the element to where pointer is directed.
+ */
+final case class ShredModelEntry(ptr: SchemaPointer, subSchema: Schema, isNotNullOverride: Boolean) {
+
+  /**
+   * columnName, nullability, columnType and compressionEncoding are used for SQL statement definition of corresponding
+   * redshift column.
+   */
+  lazy val columnName: String = ptr.getName
+
+  lazy val isNullable: Boolean = isNotNullOverride || subSchema.canBeNull
+
+  lazy val columnType: ShredModelEntry.ColumnType = columnTypeSuggestions
+    .find(_.apply(subSchema).isDefined)
+    .flatMap(_.apply(subSchema))
+    .getOrElse(ShredModelEntry.ColumnType.RedshiftVarchar(ShredModelEntry.VARCHAR_SIZE))
+
+  lazy val compressionEncoding: ShredModelEntry.CompressionEncoding = (subSchema.`enum`, columnType) match {
+    case (Some(_), ShredModelEntry.ColumnType.RedshiftVarchar(size)) if size <= 255 =>
+      ShredModelEntry.CompressionEncoding.Text255Encoding
+    case (_, ShredModelEntry.ColumnType.RedshiftBoolean) => ShredModelEntry.CompressionEncoding.RunLengthEncoding
+    case (_, ShredModelEntry.ColumnType.RedshiftDouble) => ShredModelEntry.CompressionEncoding.RawEncoding
+    case _ => ShredModelEntry.CompressionEncoding.ZstdEncoding
+  }
+
+  /**
+   * Extract the string representation of this entry from the event body. Factory relies on the validation done by the 
```

Comment:
> Factory relies on the validation done by the enrich

For some reason this is making me nervous, but maybe I'm too paranoid.  It just feels like we've got 95% of the way to having a robust transformer/loader, but then we leave a chance for something unexpected to happen if the loader disagrees with enrich about a schema.

Help me work out what could go wrong.  I suppose if the schema gets patched in production then loading would break.  Is there any other scenario where something could go wrong?

I'm not saying to change anything.... but how much extra work would it be to use the schema in `stringFactory`?

Reply:
Change is fairly small
- truncate the strings (currently still done in the trasformer, since it truncates atomic values too)
- write bunch of regexes to validate the strings for each type.

I am concerned about performance regression it will cause.

Validation here won't save us from schema patching, as warehouse types could be anything.

Reply:
as for the use cases I know only one
- large negative numbers break // maximum 30 -> SMALLINT, works for all numbers